crc/services/lookup_service.py

```diff
@@ -149,39 +149,40 @@ def build_lookup_table(data_model: FileDataModel, value_column, label_column, wo
             lookup_data = LookupDataModel(lookup_file_model=lookup_model,
                                           value=row[value_column],
                                           label=row[label_column],
-                                          data=row.to_json())
+                                          data=row.to_dict())
             db.session.add(lookup_data)
         db.session.commit()
         return lookup_model
 
     @staticmethod
-    def _run_lookup_query(lookup_file_model, query, limit):
+    def _run_lookup_query(lookup_file_model, query, value, limit):
         db_query = LookupDataModel.query.filter(LookupDataModel.lookup_file_model == lookup_file_model)
+        if value is not None:  # Then just find the model with that value
+            db_query = db_query.filter(LookupDataModel.value == value)
+        else:
+            # Build a full text query that takes all the terms provided and executes each term as a prefix query, and
+            # OR's those queries together.  The order of the results is handled as a standard "Like" on the original
+            # string which seems to work intuitively for most entries.
+            query = re.sub('[^A-Za-z0-9 ]+', '', query)  # Strip out non ascii characters.
+            query = re.sub(' {2,}', ' ', query)  # Convert multiple spaces to just one space, as we split on spaces.
```

Comment:
Need to add the `r` before the regex strings. Also, using `\s+` will catch also line breaks, carriage returns, and tab characters.
```
query = re.sub(r'[^A-Za-z0-9 ]+', '', query)  # Strip out non ascii characters.
query = re.sub(r'\s+', ' ', query)  # Convert multiple spaces to just one space, as we split on spaces.
```